app/controllers/divisions_controller.rb

```diff
@@ -27,19 +27,14 @@ def index
       @sort = params[:sort]
       @rdisplay = params[:rdisplay]
       @house = params[:house] unless params[:house] == "all"
-      if params[:date] =~ /^\d{4}$/
-        @year = params[:date]
-      elsif params[:date] =~/^\d{4}-\d{2}$/
-        @month = params[:date]
-      elsif params[:date]
-        begin
-          @date = Date.parse(params[:date])
-        rescue ArgumentError => e
-          render 'home/error_404', status: 404
-        end
+
+      begin
+        @date_start, @date_end, @date_range = DivisionParameterParser.get_date_range(params[:date]) if params[:date]
+        @date_start, @date_end, @date_range = DivisionParameterParser.get_date_range(@years.last.to_s) if @date_start.nil? && @date_end.nil? && @rdisplay.nil?
+      rescue ArgumentError => e
```

Comment:
I know this is just carried over from above but I've realised we don't use the assigned `e` variable so we can make this just:

```ruby
rescue ArgumentError
```